packages/opencensus-core/src/tags/propagation/binary-serializer.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// import {TextDecoder} from 'util';
+import * as VarInt from '../../internal/VarInt';
+import {TagMap} from '../tag-map';
+import {TagKey, TagValue} from '../types';
+
+export let VERSION_ID = 0;
+export let TAG_FIELD_ID = 0;
+// This size limit only applies to the bytes representing tag keys and values.
+const TAG_MAP_SERIALIZED_SIZE_LIMIT = 8192;
+
+// Serializes a TagMap to the on-the-wire format.
+export function serializeBinary(tagMap: TagMap): Int8Array {
+  let totalCharacters = 0;
+  const arr: Int8Array = new Int8Array([VERSION_ID]);
```

Comment:
The Node gRPC library expects a ```Buffer``` for binary headers if the normalized key ends with '-bin', in this case our key name is [**"grpc-tags-bin"**](https://github.com/grpc/grpc-java/blob/9a38dea91c6194f7a882e2be1c2846567e9cdcda/core/src/main/java/io/grpc/internal/CensusStatsModule.java#L116). I think we should use ```Buffer``` here. Similar to https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-propagation-binaryformat/src/binary-format.ts#L53

Reply:
What should the size be allocated initially? The link given here [max-size](https://github.com/census-instrumentation/opencensus-specs/blob/master/encodings/BinaryEncoding.md#tag-context)
broken.